### ISSUE TITLE
Activity Form: General Stationary Combustion (GSC) excluding line tracing 

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-reporting#253]
#### Description:
As an industrial reporter, I want to enter general stationary combustion (GSC) emissions in my report, so that I can fulfill my reporting obligations

blocked by #256 

[Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=7837-20066&t=WU1VPKoUDpbR5KCv-4): 

This is the first form we implement fully following the minispike
It is meant to also test how to handle the uiSchema and field ordering

This is about generating the form from what the API returns

Brief description of the work involved, including any user research context or key decisions.

Saving the data entered in this form is not a requirement for this ticket as we don't have the data structures defined yet for how we're going to store our reporting data in the database.

#### Acceptance Criteria:

Given I am on the general stationary combustion activity form page
When I first land on the page
Then the base schema (non-conditional) parts of the form are displayed

Given the base schema has rendered correctly
When I enter a gas type
Then the fields that are conditional on gas type are dynamically added to the form

Given the gas type data has been entered
When I enter a methodology
Then the fields that are conditional on methodoology are dynamically added to the form

#### Development Checklist:

- [x] GSC page fetches the base schema from the API service on initial load
- [x] Conditional gas type fields render dynamically
- [x] Conditional methodology fields render dynamically
- [x] Options for how to apply the UI Schema & where to store it have been explored & documented
- [ ] Matches the wireframe
- [ ] Tested with Vitest
- [ ] Documentation on how to generate a form from a configuration

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocking 
#271 
#272 

#### Notes:

-

**Comment by pbastia** [comment]
@patriciarussellCAS 👀

**Comment by patriciarussellCAS** [comment]
Flagging that we are not including the "allocation of emissions" section - this is still being confirmed by BA.

**Comment by dleard** [comment]
ui schema will need to live in the database

**Comment by dleard** [comment]
This can be tested in dev by:
logging in via the shared dashboard with a bceid
going to /reporting/activities/general_stationary_combustion directly via the address bar (no way currently to get there by clicking through the app)

**Comment by patriciarussellCAS** [comment]
I tried reviewing this page and got the following error: Error
An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.

May need someone else to review this while I'm away.

**Comment by dleard** [comment]
> I tried reviewing this page and got the following error: Error An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.
> 
> May need someone else to review this while I'm away.

Ah yeah, I see what's happening here. The schema data that's supposed to be added in the migration is being wiped out by a load-data job at the end. I've added a PR to fix this. Thanks to @joshgamache for pointing out the issue last week

**Comment by dleard** [comment]
@patriciarussellCAS the link to the page has changed since the comment above, it is now:
/reporting/activities/gsc_excluding_line_tracing

**Comment by patriciarussellCAS** [comment]
Thanks @dleard - I think this is good to close now. Do we need to fix the task list on the left so it no longer lists facilities? Or will that be done through the task list ticket?

**Comment by dleard** [comment]
The task list is just a stand-in for aesthetics at the moment (it does nothing), it will be worked on in another ticket
